__tests__/__snapshots__/runner.js.snap

```diff
@@ -2464,6 +2468,268 @@ export class UsersApi {
 export default UsersApi;"
 `;
 
+exports[`Should resolve "additionalProperties" 1`] = `
+"// tslint:disable
+
+import * as request from \\"superagent\\";
+import {
+    SuperAgentStatic,
+    SuperAgentRequest,
+    Response
+} from \\"superagent\\";
+
+export type RequestHeaders = {
+    [header: string]: string;
+}
+export type RequestHeadersHandler = (headers: RequestHeaders) => RequestHeaders;
+
+export type ConfigureAgentHandler = (agent: SuperAgentStatic) => SuperAgentStatic;
+
+export type ConfigureRequestHandler = (agent: SuperAgentRequest) => SuperAgentRequest;
+
+export type CallbackHandler = (err: any, res ? : request.Response) => void;
+
+export type some_def = {
+    'some_def' ? : string;
+};
+
+export type test_add_props_01 = {
+    'some_prop' ? : string;
+    [key: string]: any;
+};
+
+export type test_add_props_02 = {
+    'some_prop' ? : string;
+};
+
+export type test_add_props_03 = {
+    'some_prop' ? : string;
+    [key: string]: any;
+};
+
+export type test_add_props_04 = {
+    'some_prop' ? : string;
+    [key: string]: string;
+};
+
+export type test_add_props_05 = {
+    'some_prop' ? : string;
+    [key: string]: {
+        'nested_prop' ? : string;
+        [key: string]: any;
+    };
+};
+
+export type test_add_props_06 = {
+    'some_prop' ? : string;
+    [key: string]: some_def;
+};
+
+export type test_add_props_07 = {
+
+    [key: string]: any;
+};
+
+export type test_add_props_08 = {};
+
+export type test_add_props_09 = {
+
+    [key: string]: any;
+};
+
+export type test_add_props_10 = {
+
+    [key: string]: string;
+};
+
+export type test_add_props_11 = {
+
+    [key: string]: {
+        'nested_prop' ? : string;
+        [key: string]: any;
+    };
+};
+
+export type test_add_props_12 = {
+
+    [key: string]: some_def;
+};
```

Comment:
the types `test_add_props_*` have been validated according to the description about how to handle the different cases for `additionalProperties` ...